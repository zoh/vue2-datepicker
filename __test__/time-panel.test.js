import { mount } from '@vue/test-utils';
import TimePanel from '../src/time/time-panel';
import ListColumns from '../src/time/list-columns.vue';

let wrapper;

afterEach(() => {
  wrapper.destroy();
});

describe('TimePanel', () => {
  it('render: correct classes of the columns', () => {
    wrapper = mount(TimePanel, {
      propsData: {
        value: new Date(2019, 9, 4, 12, 30, 30),
        disabledTime: date => date.getHours() < 10,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('render: correct columns by format', () => {
    wrapper = mount(TimePanel, {
      propsData: {
        value: new Date(2019, 9, 4),
        format: 'hh:mm a',
        minuteStep: 30,
        hourOptions: Array.from({ length: 10 }).map((_, i) => i + 8),
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('render: correct classes of the fixed time list', () => {
    wrapper = mount(TimePanel, {
      propsData: {
        value: new Date(2019, 10, 9, 12, 30),
        disabledTime: date => date.getHours() < 10,
        timePickerOptions: {
          start: '08:30',
          step: '00:30',
          end: '18:30',
        },
        format: 'HH:mm',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('render: correct 12hours in the fixed time list', () => {
    wrapper = mount(TimePanel, {
      propsData: {
        value: new Date(2019, 10, 9, 12, 30),
        timePickerOptions: {
          start: '08:30',
          step: '00:30',
          end: '18:30',
        },
        format: 'hh:mm A',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('feat: emit select event when click', () => {
    wrapper = mount(TimePanel, {
      propsData: {
        format: 'hh:mm:ss a',
        defaultValue: new Date(2019, 9, 10, 2),
      },
    });
    const hour = wrapper.find('[data-type=hour] li:nth-child(2)');
    hour.trigger('click');
    expect(wrapper.emitted().select[0][0]).toEqual(new Date(2019, 9, 10, 1));
    wrapper.setProps({ value: new Date(2019, 9, 10, 1) });
    const minute = wrapper.find('[data-type=minute] li:nth-child(2)');
    minute.trigger('click');
    expect(wrapper.emitted().select[1][0]).toEqual(new Date(2019, 9, 10, 1, 1));
    wrapper.setProps({ value: new Date(2019, 9, 10, 1, 1) });
    const second = wrapper.find('[data-type=second] li:nth-child(2)');
    second.trigger('click');
    expect(wrapper.emitted().select[2][0]).toEqual(new Date(2019, 9, 10, 1, 1, 1));
    wrapper.setProps({ value: new Date(2019, 9, 10, 1, 1, 1) });
    const pm = wrapper.find('[data-type=ampm] li:nth-child(2)');
    pm.trigger('click');
    expect(wrapper.emitted().select[3][0]).toEqual(new Date(2019, 9, 10, 13, 1, 1));
  });

  it('feat: disabledTime should not emit event', () => {
    wrapper = mount(TimePanel, {
      propsData: {
        value: new Date(2019, 9, 4, 12, 30, 30),
        disabledTime: date => date.getHours() < 10,
      },
    });
    const hour = wrapper.find('[data-type=hour] li:nth-child(2)');
    hour.trigger('click');
    expect(wrapper.emitted().select).toBeUndefined();
  });

  it('fix: when the custom format pass into time panel', () => {
    wrapper = mount(TimePanel, {
      propsData: {
        value: new Date(),
        format: {},
      },
    });
    const cols = wrapper.find(ListColumns);
    expect(cols.props('showHour')).toBe(true);
    expect(cols.props('showMinute')).toBe(true);
    expect(cols.props('showSecond')).toBe(true);
  });
});
