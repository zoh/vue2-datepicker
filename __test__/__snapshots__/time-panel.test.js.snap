// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimePanel render: correct 12hours in the fixed time list 1`] = `
<div
  class="mx-time"
>
  <!---->
   
  <div
    class="mx-time-content"
  >
    <div
      class="mx-scrollbar"
      style="position: relative; overflow: hidden;"
    >
      <div
        class="mx-scrollbar-wrap"
        style="margin-right: -0px;"
      >
        <div
          class="mx-time-option"
        >
          
    08:30 AM
  
        </div>
        <div
          class="mx-time-option"
        >
          
    09:00 AM
  
        </div>
        <div
          class="mx-time-option"
        >
          
    09:30 AM
  
        </div>
        <div
          class="mx-time-option"
        >
          
    10:00 AM
  
        </div>
        <div
          class="mx-time-option"
        >
          
    10:30 AM
  
        </div>
        <div
          class="mx-time-option"
        >
          
    11:00 AM
  
        </div>
        <div
          class="mx-time-option"
        >
          
    11:30 AM
  
        </div>
        <div
          class="mx-time-option"
        >
          
    12:00 PM
  
        </div>
        <div
          class="mx-time-option active"
        >
          
    12:30 PM
  
        </div>
        <div
          class="mx-time-option"
        >
          
    01:00 PM
  
        </div>
        <div
          class="mx-time-option"
        >
          
    01:30 PM
  
        </div>
        <div
          class="mx-time-option"
        >
          
    02:00 PM
  
        </div>
        <div
          class="mx-time-option"
        >
          
    02:30 PM
  
        </div>
        <div
          class="mx-time-option"
        >
          
    03:00 PM
  
        </div>
        <div
          class="mx-time-option"
        >
          
    03:30 PM
  
        </div>
        <div
          class="mx-time-option"
        >
          
    04:00 PM
  
        </div>
        <div
          class="mx-time-option"
        >
          
    04:30 PM
  
        </div>
        <div
          class="mx-time-option"
        >
          
    05:00 PM
  
        </div>
        <div
          class="mx-time-option"
        >
          
    05:30 PM
  
        </div>
        <div
          class="mx-time-option"
        >
          
    06:00 PM
  
        </div>
        <div
          class="mx-time-option"
        >
          
    06:30 PM
  
        </div>
      </div>
       
      <div
        class="mx-scrollbar-track"
      >
        <div
          class="mx-scrollbar-thumb"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`TimePanel render: correct classes of the columns 1`] = `
<div
  class="mx-time"
>
  <!---->
   
  <div
    class="mx-time-content"
  >
    <div
      class="mx-time-columns"
    >
      <div
        class="mx-scrollbar mx-time-column"
        style="position: relative; overflow: hidden;"
      >
        <div
          class="mx-scrollbar-wrap"
          style="margin-right: -0px;"
        >
          <ul
            class="mx-time-list"
            data-index="0"
            data-type="hour"
          >
            <li
              class="mx-time-item disabled"
              data-index="0"
            >
              
        00
      
            </li>
            <li
              class="mx-time-item disabled"
              data-index="1"
            >
              
        01
      
            </li>
            <li
              class="mx-time-item disabled"
              data-index="2"
            >
              
        02
      
            </li>
            <li
              class="mx-time-item disabled"
              data-index="3"
            >
              
        03
      
            </li>
            <li
              class="mx-time-item disabled"
              data-index="4"
            >
              
        04
      
            </li>
            <li
              class="mx-time-item disabled"
              data-index="5"
            >
              
        05
      
            </li>
            <li
              class="mx-time-item disabled"
              data-index="6"
            >
              
        06
      
            </li>
            <li
              class="mx-time-item disabled"
              data-index="7"
            >
              
        07
      
            </li>
            <li
              class="mx-time-item disabled"
              data-index="8"
            >
              
        08
      
            </li>
            <li
              class="mx-time-item disabled"
              data-index="9"
            >
              
        09
      
            </li>
            <li
              class="mx-time-item"
              data-index="10"
            >
              
        10
      
            </li>
            <li
              class="mx-time-item"
              data-index="11"
            >
              
        11
      
            </li>
            <li
              class="mx-time-item active"
              data-index="12"
            >
              
        12
      
            </li>
            <li
              class="mx-time-item"
              data-index="13"
            >
              
        13
      
            </li>
            <li
              class="mx-time-item"
              data-index="14"
            >
              
        14
      
            </li>
            <li
              class="mx-time-item"
              data-index="15"
            >
              
        15
      
            </li>
            <li
              class="mx-time-item"
              data-index="16"
            >
              
        16
      
            </li>
            <li
              class="mx-time-item"
              data-index="17"
            >
              
        17
      
            </li>
            <li
              class="mx-time-item"
              data-index="18"
            >
              
        18
      
            </li>
            <li
              class="mx-time-item"
              data-index="19"
            >
              
        19
      
            </li>
            <li
              class="mx-time-item"
              data-index="20"
            >
              
        20
      
            </li>
            <li
              class="mx-time-item"
              data-index="21"
            >
              
        21
      
            </li>
            <li
              class="mx-time-item"
              data-index="22"
            >
              
        22
      
            </li>
            <li
              class="mx-time-item"
              data-index="23"
            >
              
        23
      
            </li>
          </ul>
        </div>
         
        <div
          class="mx-scrollbar-track"
        >
          <div
            class="mx-scrollbar-thumb"
          />
        </div>
      </div>
      <div
        class="mx-scrollbar mx-time-column"
        style="position: relative; overflow: hidden;"
      >
        <div
          class="mx-scrollbar-wrap"
          style="margin-right: -0px;"
        >
          <ul
            class="mx-time-list"
            data-index="1"
            data-type="minute"
          >
            <li
              class="mx-time-item"
              data-index="0"
            >
              
        00
      
            </li>
            <li
              class="mx-time-item"
              data-index="1"
            >
              
        01
      
            </li>
            <li
              class="mx-time-item"
              data-index="2"
            >
              
        02
      
            </li>
            <li
              class="mx-time-item"
              data-index="3"
            >
              
        03
      
            </li>
            <li
              class="mx-time-item"
              data-index="4"
            >
              
        04
      
            </li>
            <li
              class="mx-time-item"
              data-index="5"
            >
              
        05
      
            </li>
            <li
              class="mx-time-item"
              data-index="6"
            >
              
        06
      
            </li>
            <li
              class="mx-time-item"
              data-index="7"
            >
              
        07
      
            </li>
            <li
              class="mx-time-item"
              data-index="8"
            >
              
        08
      
            </li>
            <li
              class="mx-time-item"
              data-index="9"
            >
              
        09
      
            </li>
            <li
              class="mx-time-item"
              data-index="10"
            >
              
        10
      
            </li>
            <li
              class="mx-time-item"
              data-index="11"
            >
              
        11
      
            </li>
            <li
              class="mx-time-item"
              data-index="12"
            >
              
        12
      
            </li>
            <li
              class="mx-time-item"
              data-index="13"
            >
              
        13
      
            </li>
            <li
              class="mx-time-item"
              data-index="14"
            >
              
        14
      
            </li>
            <li
              class="mx-time-item"
              data-index="15"
            >
              
        15
      
            </li>
            <li
              class="mx-time-item"
              data-index="16"
            >
              
        16
      
            </li>
            <li
              class="mx-time-item"
              data-index="17"
            >
              
        17
      
            </li>
            <li
              class="mx-time-item"
              data-index="18"
            >
              
        18
      
            </li>
            <li
              class="mx-time-item"
              data-index="19"
            >
              
        19
      
            </li>
            <li
              class="mx-time-item"
              data-index="20"
            >
              
        20
      
            </li>
            <li
              class="mx-time-item"
              data-index="21"
            >
              
        21
      
            </li>
            <li
              class="mx-time-item"
              data-index="22"
            >
              
        22
      
            </li>
            <li
              class="mx-time-item"
              data-index="23"
            >
              
        23
      
            </li>
            <li
              class="mx-time-item"
              data-index="24"
            >
              
        24
      
            </li>
            <li
              class="mx-time-item"
              data-index="25"
            >
              
        25
      
            </li>
            <li
              class="mx-time-item"
              data-index="26"
            >
              
        26
      
            </li>
            <li
              class="mx-time-item"
              data-index="27"
            >
              
        27
      
            </li>
            <li
              class="mx-time-item"
              data-index="28"
            >
              
        28
      
            </li>
            <li
              class="mx-time-item"
              data-index="29"
            >
              
        29
      
            </li>
            <li
              class="mx-time-item active"
              data-index="30"
            >
              
        30
      
            </li>
            <li
              class="mx-time-item"
              data-index="31"
            >
              
        31
      
            </li>
            <li
              class="mx-time-item"
              data-index="32"
            >
              
        32
      
            </li>
            <li
              class="mx-time-item"
              data-index="33"
            >
              
        33
      
            </li>
            <li
              class="mx-time-item"
              data-index="34"
            >
              
        34
      
            </li>
            <li
              class="mx-time-item"
              data-index="35"
            >
              
        35
      
            </li>
            <li
              class="mx-time-item"
              data-index="36"
            >
              
        36
      
            </li>
            <li
              class="mx-time-item"
              data-index="37"
            >
              
        37
      
            </li>
            <li
              class="mx-time-item"
              data-index="38"
            >
              
        38
      
            </li>
            <li
              class="mx-time-item"
              data-index="39"
            >
              
        39
      
            </li>
            <li
              class="mx-time-item"
              data-index="40"
            >
              
        40
      
            </li>
            <li
              class="mx-time-item"
              data-index="41"
            >
              
        41
      
            </li>
            <li
              class="mx-time-item"
              data-index="42"
            >
              
        42
      
            </li>
            <li
              class="mx-time-item"
              data-index="43"
            >
              
        43
      
            </li>
            <li
              class="mx-time-item"
              data-index="44"
            >
              
        44
      
            </li>
            <li
              class="mx-time-item"
              data-index="45"
            >
              
        45
      
            </li>
            <li
              class="mx-time-item"
              data-index="46"
            >
              
        46
      
            </li>
            <li
              class="mx-time-item"
              data-index="47"
            >
              
        47
      
            </li>
            <li
              class="mx-time-item"
              data-index="48"
            >
              
        48
      
            </li>
            <li
              class="mx-time-item"
              data-index="49"
            >
              
        49
      
            </li>
            <li
              class="mx-time-item"
              data-index="50"
            >
              
        50
      
            </li>
            <li
              class="mx-time-item"
              data-index="51"
            >
              
        51
      
            </li>
            <li
              class="mx-time-item"
              data-index="52"
            >
              
        52
      
            </li>
            <li
              class="mx-time-item"
              data-index="53"
            >
              
        53
      
            </li>
            <li
              class="mx-time-item"
              data-index="54"
            >
              
        54
      
            </li>
            <li
              class="mx-time-item"
              data-index="55"
            >
              
        55
      
            </li>
            <li
              class="mx-time-item"
              data-index="56"
            >
              
        56
      
            </li>
            <li
              class="mx-time-item"
              data-index="57"
            >
              
        57
      
            </li>
            <li
              class="mx-time-item"
              data-index="58"
            >
              
        58
      
            </li>
            <li
              class="mx-time-item"
              data-index="59"
            >
              
        59
      
            </li>
          </ul>
        </div>
         
        <div
          class="mx-scrollbar-track"
        >
          <div
            class="mx-scrollbar-thumb"
          />
        </div>
      </div>
      <div
        class="mx-scrollbar mx-time-column"
        style="position: relative; overflow: hidden;"
      >
        <div
          class="mx-scrollbar-wrap"
          style="margin-right: -0px;"
        >
          <ul
            class="mx-time-list"
            data-index="2"
            data-type="second"
          >
            <li
              class="mx-time-item"
              data-index="0"
            >
              
        00
      
            </li>
            <li
              class="mx-time-item"
              data-index="1"
            >
              
        01
      
            </li>
            <li
              class="mx-time-item"
              data-index="2"
            >
              
        02
      
            </li>
            <li
              class="mx-time-item"
              data-index="3"
            >
              
        03
      
            </li>
            <li
              class="mx-time-item"
              data-index="4"
            >
              
        04
      
            </li>
            <li
              class="mx-time-item"
              data-index="5"
            >
              
        05
      
            </li>
            <li
              class="mx-time-item"
              data-index="6"
            >
              
        06
      
            </li>
            <li
              class="mx-time-item"
              data-index="7"
            >
              
        07
      
            </li>
            <li
              class="mx-time-item"
              data-index="8"
            >
              
        08
      
            </li>
            <li
              class="mx-time-item"
              data-index="9"
            >
              
        09
      
            </li>
            <li
              class="mx-time-item"
              data-index="10"
            >
              
        10
      
            </li>
            <li
              class="mx-time-item"
              data-index="11"
            >
              
        11
      
            </li>
            <li
              class="mx-time-item"
              data-index="12"
            >
              
        12
      
            </li>
            <li
              class="mx-time-item"
              data-index="13"
            >
              
        13
      
            </li>
            <li
              class="mx-time-item"
              data-index="14"
            >
              
        14
      
            </li>
            <li
              class="mx-time-item"
              data-index="15"
            >
              
        15
      
            </li>
            <li
              class="mx-time-item"
              data-index="16"
            >
              
        16
      
            </li>
            <li
              class="mx-time-item"
              data-index="17"
            >
              
        17
      
            </li>
            <li
              class="mx-time-item"
              data-index="18"
            >
              
        18
      
            </li>
            <li
              class="mx-time-item"
              data-index="19"
            >
              
        19
      
            </li>
            <li
              class="mx-time-item"
              data-index="20"
            >
              
        20
      
            </li>
            <li
              class="mx-time-item"
              data-index="21"
            >
              
        21
      
            </li>
            <li
              class="mx-time-item"
              data-index="22"
            >
              
        22
      
            </li>
            <li
              class="mx-time-item"
              data-index="23"
            >
              
        23
      
            </li>
            <li
              class="mx-time-item"
              data-index="24"
            >
              
        24
      
            </li>
            <li
              class="mx-time-item"
              data-index="25"
            >
              
        25
      
            </li>
            <li
              class="mx-time-item"
              data-index="26"
            >
              
        26
      
            </li>
            <li
              class="mx-time-item"
              data-index="27"
            >
              
        27
      
            </li>
            <li
              class="mx-time-item"
              data-index="28"
            >
              
        28
      
            </li>
            <li
              class="mx-time-item"
              data-index="29"
            >
              
        29
      
            </li>
            <li
              class="mx-time-item active"
              data-index="30"
            >
              
        30
      
            </li>
            <li
              class="mx-time-item"
              data-index="31"
            >
              
        31
      
            </li>
            <li
              class="mx-time-item"
              data-index="32"
            >
              
        32
      
            </li>
            <li
              class="mx-time-item"
              data-index="33"
            >
              
        33
      
            </li>
            <li
              class="mx-time-item"
              data-index="34"
            >
              
        34
      
            </li>
            <li
              class="mx-time-item"
              data-index="35"
            >
              
        35
      
            </li>
            <li
              class="mx-time-item"
              data-index="36"
            >
              
        36
      
            </li>
            <li
              class="mx-time-item"
              data-index="37"
            >
              
        37
      
            </li>
            <li
              class="mx-time-item"
              data-index="38"
            >
              
        38
      
            </li>
            <li
              class="mx-time-item"
              data-index="39"
            >
              
        39
      
            </li>
            <li
              class="mx-time-item"
              data-index="40"
            >
              
        40
      
            </li>
            <li
              class="mx-time-item"
              data-index="41"
            >
              
        41
      
            </li>
            <li
              class="mx-time-item"
              data-index="42"
            >
              
        42
      
            </li>
            <li
              class="mx-time-item"
              data-index="43"
            >
              
        43
      
            </li>
            <li
              class="mx-time-item"
              data-index="44"
            >
              
        44
      
            </li>
            <li
              class="mx-time-item"
              data-index="45"
            >
              
        45
      
            </li>
            <li
              class="mx-time-item"
              data-index="46"
            >
              
        46
      
            </li>
            <li
              class="mx-time-item"
              data-index="47"
            >
              
        47
      
            </li>
            <li
              class="mx-time-item"
              data-index="48"
            >
              
        48
      
            </li>
            <li
              class="mx-time-item"
              data-index="49"
            >
              
        49
      
            </li>
            <li
              class="mx-time-item"
              data-index="50"
            >
              
        50
      
            </li>
            <li
              class="mx-time-item"
              data-index="51"
            >
              
        51
      
            </li>
            <li
              class="mx-time-item"
              data-index="52"
            >
              
        52
      
            </li>
            <li
              class="mx-time-item"
              data-index="53"
            >
              
        53
      
            </li>
            <li
              class="mx-time-item"
              data-index="54"
            >
              
        54
      
            </li>
            <li
              class="mx-time-item"
              data-index="55"
            >
              
        55
      
            </li>
            <li
              class="mx-time-item"
              data-index="56"
            >
              
        56
      
            </li>
            <li
              class="mx-time-item"
              data-index="57"
            >
              
        57
      
            </li>
            <li
              class="mx-time-item"
              data-index="58"
            >
              
        58
      
            </li>
            <li
              class="mx-time-item"
              data-index="59"
            >
              
        59
      
            </li>
          </ul>
        </div>
         
        <div
          class="mx-scrollbar-track"
        >
          <div
            class="mx-scrollbar-thumb"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`TimePanel render: correct classes of the fixed time list 1`] = `
<div
  class="mx-time"
>
  <!---->
   
  <div
    class="mx-time-content"
  >
    <div
      class="mx-scrollbar"
      style="position: relative; overflow: hidden;"
    >
      <div
        class="mx-scrollbar-wrap"
        style="margin-right: -0px;"
      >
        <div
          class="mx-time-option disabled"
        >
          
    08:30
  
        </div>
        <div
          class="mx-time-option disabled"
        >
          
    09:00
  
        </div>
        <div
          class="mx-time-option disabled"
        >
          
    09:30
  
        </div>
        <div
          class="mx-time-option"
        >
          
    10:00
  
        </div>
        <div
          class="mx-time-option"
        >
          
    10:30
  
        </div>
        <div
          class="mx-time-option"
        >
          
    11:00
  
        </div>
        <div
          class="mx-time-option"
        >
          
    11:30
  
        </div>
        <div
          class="mx-time-option"
        >
          
    12:00
  
        </div>
        <div
          class="mx-time-option active"
        >
          
    12:30
  
        </div>
        <div
          class="mx-time-option"
        >
          
    13:00
  
        </div>
        <div
          class="mx-time-option"
        >
          
    13:30
  
        </div>
        <div
          class="mx-time-option"
        >
          
    14:00
  
        </div>
        <div
          class="mx-time-option"
        >
          
    14:30
  
        </div>
        <div
          class="mx-time-option"
        >
          
    15:00
  
        </div>
        <div
          class="mx-time-option"
        >
          
    15:30
  
        </div>
        <div
          class="mx-time-option"
        >
          
    16:00
  
        </div>
        <div
          class="mx-time-option"
        >
          
    16:30
  
        </div>
        <div
          class="mx-time-option"
        >
          
    17:00
  
        </div>
        <div
          class="mx-time-option"
        >
          
    17:30
  
        </div>
        <div
          class="mx-time-option"
        >
          
    18:00
  
        </div>
        <div
          class="mx-time-option"
        >
          
    18:30
  
        </div>
      </div>
       
      <div
        class="mx-scrollbar-track"
      >
        <div
          class="mx-scrollbar-thumb"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`TimePanel render: correct columns by format 1`] = `
<div
  class="mx-time"
>
  <!---->
   
  <div
    class="mx-time-content"
  >
    <div
      class="mx-time-columns"
    >
      <div
        class="mx-scrollbar mx-time-column"
        style="position: relative; overflow: hidden;"
      >
        <div
          class="mx-scrollbar-wrap"
          style="margin-right: -0px;"
        >
          <ul
            class="mx-time-list"
            data-index="0"
            data-type="hour"
          >
            <li
              class="mx-time-item"
              data-index="0"
            >
              
        08
      
            </li>
            <li
              class="mx-time-item"
              data-index="1"
            >
              
        09
      
            </li>
            <li
              class="mx-time-item"
              data-index="2"
            >
              
        10
      
            </li>
            <li
              class="mx-time-item"
              data-index="3"
            >
              
        11
      
            </li>
          </ul>
        </div>
         
        <div
          class="mx-scrollbar-track"
        >
          <div
            class="mx-scrollbar-thumb"
          />
        </div>
      </div>
      <div
        class="mx-scrollbar mx-time-column"
        style="position: relative; overflow: hidden;"
      >
        <div
          class="mx-scrollbar-wrap"
          style="margin-right: -0px;"
        >
          <ul
            class="mx-time-list"
            data-index="1"
            data-type="minute"
          >
            <li
              class="mx-time-item active"
              data-index="0"
            >
              
        00
      
            </li>
            <li
              class="mx-time-item"
              data-index="1"
            >
              
        30
      
            </li>
          </ul>
        </div>
         
        <div
          class="mx-scrollbar-track"
        >
          <div
            class="mx-scrollbar-thumb"
          />
        </div>
      </div>
      <div
        class="mx-scrollbar mx-time-column"
        style="position: relative; overflow: hidden;"
      >
        <div
          class="mx-scrollbar-wrap"
          style="margin-right: -0px;"
        >
          <ul
            class="mx-time-list"
            data-index="2"
            data-type="ampm"
          >
            <li
              class="mx-time-item active"
              data-index="0"
            >
              
        AM
      
            </li>
            <li
              class="mx-time-item"
              data-index="1"
            >
              
        PM
      
            </li>
          </ul>
        </div>
         
        <div
          class="mx-scrollbar-track"
        >
          <div
            class="mx-scrollbar-thumb"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
